Formula/n/netatalk.rb

```diff
@@ -0,0 +1,56 @@
+class Netatalk < Formula
+  desc "File server for Macs, compliant with Apple Filing Protocol (AFP)"
+  homepage "https://netatalk.io"
+  url "https://github.com/Netatalk/netatalk/releases/download/netatalk-4-0-8/netatalk-4.0.8.tar.xz"
+  sha256 "d09d591521b27b91b1c2a6255a2e059af8c9fda43570f983f0a145706e5f7628"
+  license all_of: [
+    "GPL-2.0-only",
+    "GPL-2.0-or-later",
+    "LGPL-2.0-only",
+    "LGPL-2.1-or-later",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "MIT",
+  ]
+  head "https://github.com/Netatalk/netatalk.git", branch: "main"
+
+  depends_on "docbook-xsl" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkgconf" => :build
+
+  depends_on "berkeley-db"           # macOS bdb library is outdated
+  depends_on "libevent"
+  depends_on "libgcrypt"
+  depends_on "mysql"
+  depends_on "openldap"              # macOS LDAP.Framework is deprecated
+
+  uses_from_macos "libxslt" => :build
+  uses_from_macos "perl"
+
+  def install
+    prefix_path = ENV["HOMEBREW_PREFIX"]
+
+    args = %w[
+      -Dwith-afpstats=false
+      -Dwith-init-style=none
+      -Dwith-install-hooks=false
+      -Dwith-spotlight=false
+    ]
+
+    args << "-Dwith-bdb-path=#{prefix_path}/opt/berkeley-db"
+    args << "-Dwith-docbook-path=#{prefix_path}/opt/docbook-xsl/docbook-xsl"
+    args << "-Dwith-pam-config-path=#{prefix_path}/etc/pam.d"
+
+    args << "-Dwith-appletalk=true" if OS.linux?
+
+    system "meson", "setup", "build", *args, *std_meson_args
+    system "meson", "compile", "-C", "build", "--verbose"
+    system "meson", "install", "-C", "build"
+  end
+
+  test do
+    system sbin/"netatalk", "-V"
+    system sbin/"afpd", "-V"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
One idea would be to start the `netatalk` daemon, sleep for a second to let is stabilize, then run `asip-status localhost`. This would validate that the server daemon can spin up all child processes, and respond to a TCP request.

Main challenge is that the `netatalk` daemon must be run with root privileges. I tried using `require_root true` with no success, so a test block is clearly very different from a service block... Any suggestions here?

Reply:
I suggest running it without root privileges and checking that you get the expected error message/exit code.

Reply:
Alright, I added a line for this test now.

The way the `netatalk` daemon behaves, is that it will start up without root privileges and launch the child processes, but it won't have the privileges to use the TCP transport layer. The expected exit code is 0 for success. (Perhaps not the ideal behavior, but it is what we have right now.)